packages/itwinui-react/src/utils/components/OverflowContainer.tsx

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useMergedRefs } from '../hooks/useMergedRefs.js';
+import type { PolymorphicForwardRefComponent } from '../props.js';
+import { Box } from './Box.js';
+import { useOverflow } from '../hooks/useOverflow.js';
+import { useSafeContext } from '../hooks/useSafeContext.js';
+
+type OverflowContainerProps = {
+  /**
+   * The orientation of the overflow in container.
+   * @default 'horizontal'
+   */
+  overflowOrientation?: 'horizontal' | 'vertical';
+  itemsCount: number;
```

Comment:
This could use some JSDocs. The prop name is not self-explanatory (I think it represents the *original* items count?).